packages/core/src/utilities/cacheUtils.ts

```diff
@@ -0,0 +1,107 @@
+import cache, { ImageVolume } from '../cache';
+import { Events } from '../enums';
+import eventTarget from '../eventTarget';
+import { getConfiguration, getShouldUseSharedArrayBuffer } from '../init';
+
+/**
+ * This function will check if the cache optimization is enabled and if it is
+ * it will check if the created volume was derived from an already cached stack
+ * of images, if so it will go back to the image cache and create a view at the
+ * correct offset of the bigger volume array buffer, this will save memory.
+ *
+ * @param volumeId - The volumeId that will be checked for cache optimization
+ */
+export function setupCacheOptimizationEventListener(volumeId) {
+  const { enableCacheOptimization } = getConfiguration();
+  const shouldUseSAB = getShouldUseSharedArrayBuffer();
+
+  const performOptimization = enableCacheOptimization && shouldUseSAB;
+  if (!performOptimization) {
+    return;
+  }
+
+  eventTarget.addEventListenerOnce(
+    Events.IMAGE_VOLUME_LOADING_COMPLETED,
+    (evt) => {
+      if (evt.detail.volumeId !== volumeId) {
+        return;
+      }
+
+      const volume = cache.getVolume(volumeId);
+
+      performCacheOptimizationForVolume(volume);
+    }
+  );
+}
+
+/**
+ * Performs cache optimization for a volume by replacing the pixel data of each image
+ * in the image cache (if found) with a view of the volume's scalar data.
+ * @param options - The options for cache optimization.
+ * @param options.volumeId - The ID of the volume.
+ */
+export function performCacheOptimizationForVolume(volume) {
+  if (!(volume instanceof ImageVolume)) {
+    return;
+  }
+
+  const scalarData = volume.getScalarData();
+
+  volume.imageCacheOffsetMap.size > 0
+    ? _processImageCacheOffsetMap(volume, scalarData)
+    : _processVolumeImages(volume, scalarData);
+
+  console.log(
```

Comment:
Remove - at least for commit.  Really we need to add a log utility so that we can leave debug information in without hitting it all the time.